src/clinical/service-worker-thread/tasks.ts

```diff
@@ -137,7 +148,25 @@ const mapEntityDocuments = (
 	}
 
 	const totalDocs = entityName === ClinicalEntitySchemaNames.DONOR ? donorCount : results.length;
-	let records = results.sort(sortDocs(sort, entityName, completionStats));
+
+	let records = results;
+
+	switch (sortType) {
+		case ClinicalDataSortTypes.defaultDonor:
+			records = results.sort(sortDocs(sort, completionStats, sortDonorRecordsByCompletion));
+		case ClinicalDataSortTypes.invalidEntity:
+			const entityErrors = errors.filter((errorRecord) => errorRecord.entityName === entityName);
+			const donorIdsWithErrors = entityErrors.reduce<Set<number>>(
+				(acc, error) => acc.add(error.donorId),
+				new Set(),
+			);
```

Comment:
```js
const donorIdsWithErrors = new Set(entityErrors.map(error => error.donorId))
```
minor suggestion - use map with `Set` constructor. using `reduce` and not reducing to a value is confusing.

Reply:
Tested and Set prevents duplicate values using this syntax so it works just as well as using reduce/add